ServiceControl Integration added additional events (#5289)
Particular/docs.particular.net: samples/servicecontrol/azure-monitor-events/sample.md

---
title: Monitor with ServiceControl events with Azure Application Insights
summary: A sample showing how to monitor events published by ServiceControl with Azure Application Insights
component: ServiceControlContracts
reviewed: 2021-04-16
related:
 - servicecontrol
 - servicecontrol/contracts
 - servicecontrol/plugins
 - samples/servicecontrol/monitoring3rdparty
---

> This sample shows how to monitor activities performed in ServiceControl and ServicePulse using the integration events. The sample uses the [Learning Transport](/transports/learning/) and a portable version of the Particular Service Platform tools. Installing ServiceControl is **not** required.

Author: ```suggestion
This sample shows how to monitor a running NServiceBus system with ServiceControl and ServicePulse, as well as how to integrate with existing monitoring solutions. The sample uses the [Learning Transport](/transports/learning/) and a portable version of the Particular Service Platform tools. Installing ServiceControl is **not** required.
```

include: platformlauncher-windows-required

downloadbutton

## Running the project

Running the project will result in 3 console windows:

1. **NServiceBusEndpoint**: The endpoint that represents the system being monitored.
1. **AzureMonitorConnector**: The endpoint that subscribes to ServiceControl notification events and pushes them to Application Insights as custom telemetry events.
1. **PlatformLauncher**: Runs an in-process version of ServiceControl and ServicePulse. When the ServiceControl instance is ready, a browser window will be launched displaying the ServicePulse Dashboard.

The samples enables triggering two types of events:

> ### MessageFailed event

Author: ```suggestion
### Message Failures
```

A `MessageFailed` event is emitted when processing a message fails and the message is moved to the error queue.

To observe this in action, press <kbd>Enter</kbd> in the `NServiceBusEndpoint`console window. It will cause generate new `SimpleMessage` event that causes failure when processed.

NOTE: The exception will cause the debugger to enter a breakpoint. It may be preferable to detach the debugger in order to better observe what's going on.

When a `MessageFailed` event is received, the `AzureMonitorConnector` prints the following message in its console window: 

```
> Received ServiceControl 'MessageFailed' event for a SimpleMessage with ID 42f25e40-a673-61f3-a505-c8dee6d16f8a
```

The failed message can also be viewed in the ServicePulse browser window. Navigating to the failed message allows viewing more details about the message failure.

> ### HeartbeatStopped and HeartbeatRestored events

Author: ```suggestion
### Heartbeat statuses
```
